kasa/cli.py

```diff
@@ -309,8 +347,16 @@ async def join(dev: SmartDevice, ssid, password, keytype):
     is_flag=True,
     help="Print out discovered unsupported devices",
 )
+@click.option(
+    "--show-discovery-results",
+    envvar="KASA_SHOW_DISCOVERY_RESULTS",
+    required=False,
+    default=False,
+    is_flag=True,
+    help="Print out discovery results",
+)
 @click.pass_context
-async def discover(ctx, timeout, show_unsupported):
+async def discover(ctx, timeout, show_unsupported, show_discovery_results):
```

Comment:
I think we should consider adding a `--verbose` for all types of potentially interesting but not debug worthy information?

Instead of `--show-unsupported-devices --show-discover-results`, passing `--verbose` (or `-v`) would print out both of these? `--debug` is more developer-oriented so much of its output is not that useful for users, where this type of information could be.

Reply:
That does make sense although the `--show-discover-results` is actually restrictive, i.e. it only shows the discovery results and does not connect and do an `update()`.  This is useful for testing that the devices report the minimal attributes without having to do an update: `host`, `port`, `mac`, `model`, `alias` (needed for HA).

I could replace `--show-unsupported-device` with `--verbose` that will show discovery data +update data and rename `--show-discovery-results` to `--connect/--no-connect` with `--connect` as the default.

Reply:
I don't think we should add another "--connect/--no-connect" option to keep it simple...

 Now that I think about this, it's only about the cli tool. I'm wondering if we should even hide it behind `show_unsupported` here but rather keep it simple and print out that information always. Basically, anyone seeing the message about unsupported who's going to report an issue or wants to work figuring out to add support for that will now need to re-run the discovery again. Basically, I don't see the value of hiding these results from the cli output..

I'd bet that the regular use case is to find some devices, and having some "== Authentication failed for device ==" (just saw this first time, looks really nice btw!) entries won't hopefully ruin anyone's day :-)

Reply:
Ok will remove `--show-unsupported` and do as default.  Omitting creds will do the same as `--show-discovery-results`

Reply:
Done.  I kept --verbose which will show all the discovery responses regardless of authentication state if provided